infra/terraform/test-org/github/resources/lint.yaml

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'lint'
+
+on:
+  pull_request:
+    branches:
+      - 'master'
+      - 'main'
+
```

Comment:
Could we add make this into a tf template and allow passing in env vars? I saw a few repos which pass in custom exclude dirs, disable TF lint etc

https://github.com/terraform-google-modules/terraform-example-foundation/blob/e6089046d93be6a397509ae63b40ff78037a645a/build/lint.cloudbuild.yaml#L21
https://github.com/terraform-google-modules/terraform-google-gcloud/blob/84847e60a758f3d490e84a3efcf20b7a6a73955e/build/lint.cloudbuild.yaml#L20

Reply:
@bharathkkb - Yes, I thought about using it for the branch using data, but ran out of time last week.

I also noticed at least one repo doesn't put the `Makefile` in `.`: https://github.com/GoogleCloudPlatform/terraform-ecommerce-microservices-on-gke. My current thought is to simply fall back to a sensible default (e.g. `1` or `1.11`), but if needed we could plumb through a version override as well.

Reply:
I've mapped through EXCLUDE_LINT_DIRS, set the branch based on the repo's default, and also if no `./Makefile` it will run with `1`.  Do we really want to provide an option to disable tf_lint?

Reply:
@apeabody Could we check `./Makefile` and `infra/Makefile`. Between those two I think 99% of cases should be covered.

Reply:
Sure, we could even do something like `$(find . -name Makefile -print -quit)`, the only risk is a repo with multiple `Makefile`, but none in the root, might match the wrong one.